modules/team_routing_rule/main.tf

```
@@ -1,16 +1,48 @@
 resource "opsgenie_team_routing_rule" "this" {
   count = module.this.enabled ? 1 : 0
 
-  name     = var.team_routing_rule.name
-  team_id  = var.team_routing_rule.team_id
-  order    = try(var.team_routing_rule.order, 0)
+  name    = var.team_routing_rule.name
+  team_id = var.team_routing_rule.team_id
+  order   = try(var.team_routing_rule.order, 5)
+
   timezone = try(var.team_routing_rule.timezone, "America/Los_Angeles")
+  dynamic "time_restriction" {
+    for_each = [for time_restriction in try([var.team_routing_rule.time_restriction], []) : time_restriction if time_restriction != null]
+
+    content {
+      # NOTE: The Opsgenie Provider appears to be inconsistent with how it uses time_restriction:
+      # `restrictions` for type `weekday-and-time-of-day`
+      # `restriction` for type `time-of-day`
+      type = try(var.team_routing_rule.time_restriction.type, "weekday-and-time-of-day")
```

maybe we need to make it more strict and don't allow missing fields?
otherwise, people will use some values and skipping the others, then the module will create something that they don't anticipate.

the same for `start_day  = try(restrictions.value.start_day, "monday")`

---

I think that could work 
```suggestion
      type = var.team_routing_rule.time_restriction.type
```

---

Something like that?